ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -261,6 +250,22 @@ public void deleteTopics(final Collection<String> topicsToDelete) {
       try {
         entry.getValue().get(30, TimeUnit.SECONDS);
       } catch (final Exception e) {
+        final Throwable rootCause = ExceptionUtils.getRootCause(e);
+
+        // Checking if the Kafka cluster has the DELETE_TOPIC_ENABLE configuration enabled requires
+        // the user to have DescribeConfigs permissions (or ACL) in Kafka. To avoid giving that
+        // unnecessary permission, we can detect it by catching the TopicDeletionDisabledException.
+        if (rootCause instanceof TopicDeletionDisabledException) {
+          // If TopicDeletionDisabledException is detected, we throw the exception immediately
+          // instead of going through the rest of the topics to delete.
+          // It is now up to the caller to ignore this exception.
+          LOG.info("Cannot delete topics since '" + DELETE_TOPIC_ENABLE + "' is false. ");
+          throw new TopicDeletionDisabledException("Topic deletion is disabled. "
```

Comment:
For me, I think the 'log and throw' is an anti-pattern. Either log, or throw, not both.  This pattern just leads to the same error being logged multiple times.

The fact that this requires logic when catching this exception to know not to double log is, IMHO, a clear indication this could be simplified to just throwing here, and just logging where its caught.

Reply:
Done